src/compose/app.ts

```diff
@@ -794,8 +795,8 @@ export class App {
 					.getNetworkGateway(constants.supervisorNetworkInterface)
 					.catch(() => '127.0.0.1'),
 				(async () => ({
-					firmware: await pathExistsOnHost('/lib/firmware'),
```

Comment:
Not sure this is an improvement, I can see utility on keeping `pathExistsOnHost`

Reply:
Also not entirely sure how this fits with the current commit.

Reply:
I can separate this into another commit, but the idea here is that all references to host path checking (path on root/boot/data/state) go through host-utils.ts and isn't split between fs-utils.ts and host-utils.ts. I can add `pathExistsOnHost` into host-utils if that works for you?

Reply:
> I can add pathExistsOnHost into host-utils if that works for you?

Yeah, that works, we'll avoid the `exists(pathOnHost(...))` as happens a lot

Reply:
I added `pathExistsOnRoot` and `pathExistsOnState` to host-utils [here](https://github.com/balena-os/balena-supervisor/pull/2120/files#diff-caf13efa686c7a4bb534ffdcb897f8028fcf67898446a4db30724ec875639405R26-R42), previously the function that would check on state would get the state path first and then pass it to `pathExistsOnHost` (root). However, I didn't want to do this here because the intent is to refer to host-utils for any path-getting or path-resolving functionality.